File: ui/src/types/review.ts
export interface Review {
  id: number
  movie_id: number
  author_name: string
  rating: number
  content: string
  created_at: string
}

export interface ReviewCreate {
  author_name: string
  rating: number
  content: string
}

export interface ReviewListResponse {
  items: Review[]
  total: number
  skip: number
  limit: number
  average_rating: number | null
}

File: app/db/models/review.py
from __future__ import annotations

import datetime as dt
from typing import TYPE_CHECKING

from sqlalchemy import DateTime, Float, ForeignKey, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base

if TYPE_CHECKING:
    from app.db.models.movies import Movie


class Review(Base):
    __tablename__ = "reviews"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    movie_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("movies.id", ondelete="CASCADE"), nullable=False, index=True
    )
    author_name: Mapped[str] = mapped_column(String(255), nullable=False)
    rating: Mapped[float] = mapped_column(Float, nullable=False)  # Individual review rating (0-10)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    created_at: Mapped[dt.datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now()
    )

    movie: Mapped[Movie] = relationship("Movie", back_populates="reviews")
